### ISSUE TITLE
Covalent docking - ModuleNotFoundError: No module named 'rdkit'

### ISSUE THREAD

**Issue by Brett628** [forlilab/Meeko#148]
OS: Linux
I created an environment with conda running python3.10.14 (someone in another issue told me that Meeko might not work with Python 12 so i went safe, not sure if I did right)

I installed in this environment all the packages needed for Meeko

While doing the test proposed in the files, I encounter this error: 

```
test_env) prova@prova-VirtualBox:~/Downloads/Meeko-release/example/covalent_docking$ mk_prepare_ligand.py    --hydrate    --rigid_macrocycles    --rigidify_bonds_smarts "C=CC=C"    --rigidify_bonds_indices 2 3    --rigidify_bonds_smarts "[CX3](=O)[NX3]"    --rigidify_bonds_indices 1 3
Traceback (most recent call last):
  File "/home/prova/.local/bin/mk_prepare_ligand.py", line 12, in <module>
    from rdkit import Chem
ModuleNotFoundError: No module named 'rdkit'
```

I tried also using different env with different versions of python but the same error appears. Can someone help me please?

**Comment by diogomart** [comment]
Looks like RDKit is not installed.

**Comment by diogomart** [comment]
We are depending on prody for covalent/tethered docking at the moment, and prody doesn't work with Python 3.12. Python 3.10 works fine.

**Comment by Brett628** [comment]
Thank you for your replies, I tried to install RDKit with conda but it
didn't work, so I installed it with pip and the installation went through.
Dana Zappaterra

Il gio 8 ago 2024, 23:12 Diogo ***@***.***> ha scritto:

**Comment by Brett628** [comment]
But still, the same error appears :(
Dana Zappaterra

Il gio 8 ago 2024, 23:17 Dana Zappaterra ***@***.***> ha scritto:

**Comment by diogomart** [comment]
Then it's not installed. Maybe you installed RDKit into a different enviroment? `pip` should install in the active environment

**Comment by diogomart** [comment]
The location of the script "/home/prova/.local/bin/mk_prepare_ligand.py" suggests you didn't follow the installation instructions.

**Comment by diogomart** [comment]
Maybe, earlier, you copied the script into ~/.local/bin, and now followed the installation instructions. Try
```
mv /home/prova/.local/bin/mk_prepare_ligand.py /home/prova/.local/bin/mk_prepare_ligand.py_BKP
```

**Comment by rwxayheee** [comment]
Hi @Brett628 
My guess is also that you installed Meeko and RDKit in different environments. If you have conda, you can create an environment and have the packages installed in one line: 
`conda create -n my_meeko_env python=3.10 numpy scipy rdkit meeko -c conda-forge`

After this, if you need to add packages to this environment by `pip` (for example prody), first do: 
`conda activate my_meeko_env`
Then: 
`pip install prody`
This way prody is installed in the activated environment, `my_meeko_env`

**Comment by Brett628** [comment]
Hello, sorry for the late update.
I followed the instructions and it worked! 

For the people that maybe are reading this, afterwards this error comes up:

`TypeError: can't multiply sequence by non-int of type 'Forward'`

And it can be solved by installing pyparsing 3.1.1 in the same environment:
`pip install pyparsing==3.1.1`

Thanks very much to @diogomart and @rwxayheee for helping me, now it's time to dock :)
